src/Compilers/CSharp/Portable/Parser/LanguageParser_InterpolatedString.cs

```diff
@@ -51,142 +34,174 @@ private ExpressionSyntax ParseInterpolatedStringToken()
             // 
             // The substitution will end up being invisible to external APIs and clients such as the IDE, as
             // they have no way to ask for the stream of tokens before parsing.
-            //
 
+            Debug.Assert(this.CurrentToken.Kind == SyntaxKind.InterpolatedStringToken);
             var originalToken = this.EatToken();
+
             var originalText = originalToken.ValueText; // this is actually the source text
             Debug.Assert(originalText[0] == '$' || originalText[0] == '@');
 
-            var isAltInterpolatedVerbatim = originalText.Length > 2 && originalText[0] == '@'; // @$
-            var isVerbatim = isAltInterpolatedVerbatim || (originalText.Length > 2 && originalText[1] == '@');
+            var isVerbatim = (originalText[0] == '$' && originalText[1] == '@') ||
+                             (originalText[0] == '@' && originalText[1] == '$');
 
-            Debug.Assert(originalToken.Kind == SyntaxKind.InterpolatedStringToken);
+            // compute the positions of the interpolations in the original string literal, if there was an error or not,
+            // and where the close quote can be found.
             var interpolations = ArrayBuilder<Lexer.Interpolation>.GetInstance();
-            SyntaxDiagnosticInfo error;
-            bool closeQuoteMissing;
-            using (var tempLexer = new Lexer(Text.SourceText.From(originalText), this.Options, allowPreprocessorDirectives: false))
+
+            rescanInterpolation(out var openQuoteRange, out var error, out var closeQuoteRange);
```

Comment:
>openQuoteRange [](http://example.com/codeflow?start=40&length=14)

Can we make any assertions that these quote ranges are within the original string?

Reply:
we can.  though we'll also horrifically crash if they're not since we slice using those ranges.  so teh assertion won't be doing much more than just precrashing :)